webrender/src/device/gfx/device.rs

```diff
@@ -1249,31 +1259,31 @@ impl<B: hal::Backend> Device<B> {
             .programs
             .get_mut(&self.bound_program)
             .expect("Program not found.");
+        let mut command_buffer = self.command_pool[self.next_id].acquire_command_buffer();
         let desc_set = self.descriptor_pools[self.next_id].get(&program.shader_kind);
+        unsafe { command_buffer.begin(); }
         for &(index, sampler_name) in SAMPLERS[0 .. 5].iter() {
             program.bind_texture(
                 &self.device,
                 desc_set,
                 &self.images[&self.bound_textures[index]].core,
                 sampler_name,
+                &mut command_buffer,
             );
         }
-        let desc_set = self.descriptor_pools_global.get(&program.shader_kind);
+        let desc_set = self.descriptor_pools_global[self.next_id].get(&program.shader_kind);
```

Comment:
if we cache the descriptor sets between frames, and treat each desc set as immutable for as long as it's tracked, we don't need separate pools per frame. So `descriptor_pools_xxx` would need to switch from being a small vector of at least 1 element to exactly a single instance of a descriptor pool.

Reply:
After this PR and https://github.com/szeged/webrender/pull/282 lands we can move forward in this direction.